### ISSUE TITLE
Add case sensitive support for adapter naming 

### ISSUE THREAD

**Issue by theharshbhatia** [hubotio/generator-hubot#84]
I tried to init using "Skype" instead of "skype" as mentioned in this list 
https://hubot.github.com/docs/adapters/
but unable to install as hubot-Skype as package is not available in npm.
I was able to init using "skype" as adapter in it.

**Comment by gr2m** [comment]
confirmed 

```
$ yo hubot

     _-----_     
    |       |    ╭──────────────────────────────────────────╮
    |--(o)--|    │ Update available: 2.0.0 (current: 1.8.5) │
   `---------´   │     Run npm install -g yo to update.     │
    ( _´U`_ )    ╰──────────────────────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

                     _____________________________  
                    /                             \ 
   //\              |      Extracting input for    |
  ////\    _____    |   self-replication process   |
 //////\  /_____\   \                             / 
 ======= |[^_/\_]|   /----------------------------  
  |   | _|___@@__|__                                
  +===+/  ///     \_\                               
   | |_\ /// HUBOT/\\                             
   |___/\//      /  \\                            
         \      /   +---+                            
          \____/    |   |                            
           | //|    +===+                            
            \//      |xx|                            

? Owner Gregor <gregor@martynus.net>
? Bot name skypehubottest
? Description A simple helpful robot for your Company
? Bot adapter Skype
   create bin/hubot
   create bin/hubot.cmd
   create Procfile
   create README.md
   create external-scripts.json
   create hubot-scripts.json
   create .gitignore
   create package.json
   create scripts/example.coffee
   create .editorconfig
                     _____________________________  
 _____              /                             \ 
 \    \             |   Self-replication process   |
 |    |    _____    |          complete...         |
 |__\\|   /_____\   \     Good luck with that.    / 
   |//+  |[^_/\_]|   /----------------------------  
  |   | _|___@@__|__                                
  +===+/  ///     \_\                               
   | |_\ /// HUBOT/\\                             
   |___/\//      /  \\                            
         \      /   +---+                            
          \____/    |   |                            
           | //|    +===+                            
            \//      |xx|                            

npm ERR! Darwin 16.6.0
npm ERR! argv "/Users/gregor/.nvm/versions/node/v6.9.5/bin/node" "/Users/gregor/.nvm/versions/node/v6.9.5/bin/npm" "install" "hubot" "hubot-scripts" "hubot-diagnostics" "hubot-help" "hubot-heroku-keepalive" "hubot-google-images" "hubot-google-translate" "hubot-pugme" "hubot-maps" "hubot-redis-brain" "hubot-rules" "hubot-shipit" "hubot-Skype" "--save"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code E404

npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/hubot-Skype
npm ERR! 404 
npm ERR! 404  'hubot-Skype' is not in the npm registry.
npm ERR! 404 Your package name is not valid, because 
npm ERR! 404  1. name can no longer contain capital letters
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/tmp/skypehubottest/npm-debug.lo
```

@hubotio/maintainers any reason not to lowercase the adapter name when looking up its package?

**Comment by technicalpickles** [comment]
https://hubot.github.com/docs/adapters/ doesn't list out adapters anymore, and suggests using npm or github to find them now.

I think this issue can be closed if we think about it as the documentation. It might be worth having an issue on the generator for better error handling of not finding the adapter given.
